### DISCUSSION TITLE
Security example in tutorial raise bad request

### DISCUSSION THREAD

**Discussion by JanhSander** [fastapi/fastapi#7247] [Questions] [answered]
* OS: Windows 10 enterprise
* FastAPI Version 0.60.1
* Python version: 3.8.5

I have followed the tutorial example:
https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/
When I try "See it in action" I get "Bad request".
![Skjermbilde](https://user-images.githubusercontent.com/63044278/90643661-1e22df00-e234-11ea-9f38-0dbdb068a90e.JPG)

**Comment by chrisjsimpson** [comment]
Look at the response in network tools (e.g. Chrome network in Devtools) the response is correct. Or use curl.

**Comment by JanhSander** [comment]
curl -X POST "http://127.0.0.1:8000/token" -H  "accept: application/json" -H  "Content-Type: application/x-www-form-urlencoded" -d "grant_type=&username=johndoe&password=secret&scope=&client_id=&client_secret="
{"detail":"There was an error parsing the body"}

Is this correct?

**Comment by chrisjsimpson** [comment, marked as answer]
@JanhSander 
Here's working example based on the docs:

With wrong password:

**Request**
```
curl -v  'http://127.0.0.1:8001/token' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Accept: application/json' --data 'grant_type=password&username=johndoe&password=badpassword'
```
**Response**
```
{"detail":"Incorrect username or password"}
```

With correct password:

**Request**
```
curl -v http://127.0.0.1:8001/token -H 'Content-Type: application/x-www-form-urlencoded' -H 'Accept: application/json' --data 'grant_type=password&username=johndoe&password=secret'

```
**Response**
```
{"access_token":"johndoe","token_type":"bearer"}
```

If you don't get the above response, then, there's perhaps an error in your code

**Comment by JanhSander** [comment]
Same error:
400 Bad Request
{
  "detail": "There was an error parsing the body"
}

I think it must be an error in the docs. The code I am running is just a copy from the docs.

**Comment by chrisjsimpson** [comment]
@JanhSander Did you resolve it? If not:
[Here is example code](https://github.com/chrisjsimpson/fastapi/blob/c93ff0071d967bc7f195439dc0b4cfcde31c4df0/app/main.py#L25) which works
Currently deployed at [this location ](https://fastapi-staging.pcpink.co.uk/docs) 
username: johndoe 
password: secret

Plug: a step-by-step guide to [deploying fastapi ](https://blog.karmacomputing.co.uk/fastapi-production-deployment-with-dokku-and-github-actions/) or use the [cookie cutter project](https://github.com/tiangolo/full-stack-fastapi-postgresql).

**Comment by JanhSander** [comment]
Yes, resolved

**Comment by tiangolo** [comment]
Thanks for the help here @chrisjsimpson ! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
